File: src/main/java/com/hedvig/backoffice/services/tickets/dto/CreateTicketDto.java
package com.hedvig.backoffice.services.tickets.dto;

import com.hedvig.backoffice.graphql.types.TicketInput;
import lombok.Value;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import java.time.LocalDate;
import java.time.LocalTime;

@Value
public class CreateTicketDto {
    private String createdBy;
    private String assignedTo;
    private String memberId;
    private String referenceId;
    @Min(0)
    @Max(1)
    private Float priority;
    private TicketType type;
    private LocalDate remindNotificationDate;
    private LocalTime remindNotificationTime;
    private String remindMessage;
    private String description;
    private TicketStatus status;

    public static CreateTicketDto fromTicketInput(TicketInput t, String createdBy) {
      return new CreateTicketDto (
        createdBy,
        t.getAssignedTo(),
        null,
        null,
        t.getPriority(),
        t.getType(),
        t.getRemindNotificationDate(),
        t.getRemindNotificationTime(),
        t.getRemindMessage(),
        t.getDescription(),
        t.getStatus()
      );
    }
}

File: src/main/java/com/hedvig/backoffice/services/tickets/dto/CreateTicketDto.kt
package com.hedvig.backoffice.services.tickets.dto

import com.hedvig.backoffice.graphql.types.TicketInput

import javax.validation.constraints.Max
import javax.validation.constraints.Min
import java.time.LocalDate
import java.time.LocalTime

data class CreateTicketDto(
    val createdBy: String,
    val assignedTo: String? = null,
    val memberId: String? = null,
    val referenceId: String? = null,
    @Min(0)
    @Max(1)
    val priority: Float? = null,
    val type: TicketType,
    val remindNotificationDate: LocalDate? = null,
    val remindNotificationTime: LocalTime? = null,
    val remindMessage: String? = null,
    val description: String? = null
) {
  companion object {
    fun from(ticketInput: TicketInput, createdBy: String): CreateTicketDto {
      return CreateTicketDto(
        createdBy = createdBy,
        assignedTo = ticketInput.assignedTo,
        priority = ticketInput.priority,
        type = ticketInput.type,
        remindNotificationDate = ticketInput.remindNotificationDate,
        remindNotificationTime = ticketInput.remindNotificationTime,
        remindMessage = ticketInput.remindMessage,
        description = ticketInput.description,
        referenceId = ticketInput.referenceId
      )
    }
  }
}
